src/models/room.js

```diff
@@ -228,6 +228,34 @@ function pendingEventsKey(roomId) {
 
 utils.inherits(Room, EventEmitter);
 
+Room.prototype.decryptCriticalEvents = function() {
```

Comment:
It would be good to have a comment describing the meaning of "critical" that we're aiming for here, so future readers know what properties to maintain.